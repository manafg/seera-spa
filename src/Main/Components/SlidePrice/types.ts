type SlidePriceProps = {
  onChange: (price: number) => void;
};

export type { SlidePriceProps };
