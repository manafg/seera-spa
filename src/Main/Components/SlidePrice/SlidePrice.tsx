import { StarIcon } from "@chakra-ui/icons";
import {
  Slider,
  SliderTrack,
  SliderFilledTrack,
  SliderThumb,
  Box,
} from "@chakra-ui/react";
import { Text } from "@chakra-ui/react";
import React from "react";

import { SlidePriceProps } from "./types";

function SlidePrice({ onChange }: SlidePriceProps) {
  return (
    <>
      <Text fontSize='lg' mb='4'>Search by price</Text>

      <Slider
        aria-label="slider-ex-4"
        onChangeEnd={onChange}
        defaultValue={200}
        min={200}
        max={350}
        step={1}
      >
        <SliderTrack bg='red.100'>
          <SliderFilledTrack bg='red.300' />
        </SliderTrack>
        <SliderThumb boxSize={6}>
          <Box color="red" as={StarIcon} />
        </SliderThumb>
      </Slider>
    </>
  );
}

export default SlidePrice;
