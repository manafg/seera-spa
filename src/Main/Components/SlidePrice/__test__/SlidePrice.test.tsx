import React from "react";

import SlidePrice from "../SlidePrice";

import "@testing-library/jest-dom/extend-expect";
import rrender from "react-test-renderer";

describe("Snapshot tests", () => {
  test("Matches existing Snapshot", () => {
    const tree = rrender.create(<SlidePrice onChange={jest.fn()} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
