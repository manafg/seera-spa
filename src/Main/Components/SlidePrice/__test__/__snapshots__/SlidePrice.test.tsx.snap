// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Matches existing Snapshot 1`] = `
Array [
  <p
    className="chakra-text css-x4lmqn"
  >
    Search by price
  </p>,
  <div
    className="chakra-slider css-snk6g7"
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "outline": 0,
        "paddingBottom": 0,
        "paddingTop": 0,
        "position": "relative",
        "touchAction": "none",
        "userSelect": "none",
      }
    }
    tabIndex={-1}
  >
    <div
      className="chakra-slider__track css-1d8jzoz"
      id="slider-track-4525779299-1"
      style={
        Object {
          "position": "absolute",
          "top": "50%",
          "transform": "translateY(-50%)",
          "width": "100%",
        }
      }
    >
      <div
        className="chakra-slider__filled-track css-5xam2"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": "50%",
            "transform": "translateY(-50%)",
            "width": "0%",
          }
        }
      />
    </div>
    <div
      aria-label="slider-ex-4"
      aria-orientation="horizontal"
      aria-valuemax={350}
      aria-valuemin={200}
      aria-valuenow={200}
      className="chakra-slider__thumb css-1wrxcqj"
      id="slider-thumb-4525779299-1"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="slider"
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitUserSelect": "none",
          "left": "calc(0% - 0px)",
          "msUserSelect": "none",
          "position": "absolute",
          "touchAction": "none",
          "userSelect": "none",
        }
      }
      tabIndex={0}
    >
      <svg
        className="chakra-icon css-mlftvw"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      type="hidden"
      value={200}
    />
  </div>,
]
`;
